pkg/multiboot/ranges.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multiboot
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/kexec"
+)
+
+var ErrNotEnoughSpace = errors.New("Not enough space")
+
+var sizeofMmap = uint32(binary.Size(Mmap{}))
+
+type Segments []kexec.Segment
+
+type rangeType string
+
+const (
+	rangeRAM      rangeType = "System RAM"
+	rangeNVACPI             = "ACPI Non-volatile Storage"
+	rangeACPI               = "ACPI Tables"
+	rangeReserved           = "Reserved"
```

Comment:
Is this meant to be an exhaustive list? See https://elixir.bootlin.com/linux/v4.14/source/arch/x86/kernel/e820.c#L995